### PR TITLE
SONARXML-7 Fix XPath rule to raise issues on file level

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java [hunk 1]
```diff
@@ -79,17 +78,31 @@ private void evaluateXPath() {
...
+      for (int i = 0; i < nodes.getLength(); i++) {
         int lineNumber = getWebSourceCode().getLineForNode(nodes.item(i));
-        if (message == null) {
-          createViolation(lineNumber, "--");
-        } else {
-          createViolation(lineNumber, message);
+        createViolation(lineNumber);
+      }
+
+    } catch (XPathExpressionException exceptionNodeSet) {
```
Comment (SonarTech) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java [hunk 2]
```diff
@@ -76,20 +74,35 @@ public String getPrefix(String uri) {
 
   private void evaluateXPath() {
     Document document = getWebSourceCode().getDocument(expression.contains(":"));
+    XPathExpression xPathExpression = getXPathExpressionForDocument(document);
 
     try {
```
Comment (lindamartin) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java]: @vilchik-elena if you're ok I will refactor this logic to have:
``` 
try {
    evaluateXPathForNodeSet(document, xPathExpression);
  } catch (XPathExpressionException exceptionNodeSet) {
    evaluateXPathForBoolean(document, xPathExpression);
  }
```

Comment (vilchik-elena) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java]: @lindamartin I'm ok, thanks!

Comment (lindamartin) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java]: @vilchik-elena Cool!

**Pull request merged**